test/runner.js

```diff
@@ -0,0 +1,185 @@
+    })
+
+  })
+
+  it('should marked broken when failing', () => {
+
+    return getResultsXML()
+      .then((results) => {
+
+        expect(results).to.have.lengthOf(1)
+        expect(results[0]['ns2:test-suite']['test-cases']).to.have.lengthOf(1)
+        
+        let isBeforeAllHookBroken = false
+
+        results[0]['ns2:test-suite']['test-cases'][0]['test-case']
+          .forEach((testCase) => {
+
+            switch(testCase.name[0]) {
+
+              case '"before" hook':
+              case '"before each" hook':
+              case '"after all" hook':
+              case '"after each" hook':
+                throw new Error('\'' + testCase.name + '\' was not expected in the results when no hook fails')
+                break
+
+              case '"before all" hook':
+                isBeforeAllHookBroken = (testCase.$.status === 'broken')
+                break;
+
+            }
+
+          })
+
+        expect(isBeforeAllHookBroken).to.be.true
+
+      })
+
+  })
+
+  it('should marked broken when failing async', () => {
+
+    return getResultsXML()
+      .then((results) => {
+
+        expect(results).to.have.lengthOf(1)
+        expect(results[0]['ns2:test-suite']['test-cases']).to.have.lengthOf(1)
+        
+        let isBeforeAllHookBroken = false
+
+        results[0]['ns2:test-suite']['test-cases'][0]['test-case']
+          .forEach((testCase) => {
+
+            switch(testCase.name[0]) {
+
+              case '"before" hook':
+              case '"before each" hook':
+              case '"after all" hook':
+              case '"after each" hook':
+                throw new Error('\'' + testCase.name + '\' was not expected in the results when no hook fails')
+                break
+
+              case '"before all" hook':
+                isBeforeAllHookBroken = (testCase.$.status === 'broken')
+                break;
+
+            }
+
+          })
+
+        expect(isBeforeAllHookBroken).to.be.true
+
+      })
+
+  })
+
+})
+
+function getResultsXML() {
+  let promises = getResults().map((result) => {
+    return new Promise((resolve, reject) => {
+      parseXmlString(result, { trim: true }, (err, xmlData) => {
+        if(err) {
+          reject(err)
+        } else {
+          resolve(xmlData);
+        }
+      })
+    })
+  })
+
+  return Promise.all(promises)
+}
+
+function getResults() {
+  return fs.readdirSync(resultsDir)
+    .filter((file) => file.endsWith('.xml'))
+    .map((file) => {
+      return fs.readFileSync(path.join(resultsDir, file))
+    })
+}
+
+function clean(done) {
+  return new Promise((resolve, reject) => {
+    rimraf(resultsDir, (err) => {
+      if(err) {
+        reject(err)
+      } else {
+        resolve()
+      }
+    })
+  })
+}
+
+function run(specs) {
+
+  disableOutput()
+  specs = specs.map((spec) => './test/fixtures/specs/' + spec + '.js')
+
+  let launcher = new Launcher(configFile, {
+    specs: specs
+  })
+
+  return launcher.run()
+    .then(enableOutput)
```

Comment:
You can also validate resulting xml here.

Look how it is done in allure-mocha for example:
https://github.com/allure-framework/mocha-allure-reporter/blob/master/test/e2e/main.spec.js